draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -148,148 +135,292 @@ capitals, as shown here.
 
 The terms 'workload' and 'container' are used interchangeably.
 
-# Recommendations {#recommendations}
+# Patterns {#patterns}
 
-This specification relies on the use of OAuth 2.0 {{RFC6749}} and
-{{RFC7523}} for client authentication using a JWT.
+## Kubernetes {#kubernetes}
 
-Service Account Tokens used in container orchestration systems are vulnerable to various threats, as outlined below:
+Kubernetes notion of identity is mainly the concept of "service accounts" which are either created explictly or automatically during deployment time. While the mean purpose of these identities is to authenticate access to the Kubernetes Control Plane it is more and more used to authenticate workloads between each other too. The credential format used to authenticate service account is JWT that are signed by the Kubernetes Control Plane.
```

Comment:
>it is more and more used to authenticate workloads between each other too

**and authenticate agains other domains such as cloud accounts.**

Reply:
I rewrote the section a bit, Kubernetes mentions the service account use cases here https://kubernetes.io/docs/concepts/security/service-accounts/#use-cases.

Before, I wrote that primary use case is to authenticate Kubernetes APIs, but I guess that's quite a old statement.

Also added references to Kubernetes docs.